vllm/v1/core/scheduler.py

```diff
@@ -227,13 +228,12 @@ def update_from_output(
         self,
         scheduler_output: "SchedulerOutput",
         model_runner_output: "ModelRunnerOutput",
-    ) -> List[Tuple[Request, int]]:
+    ) -> List[EngineCoreOutput]:
```

Comment:
@njhill do you think we should remove this method and instead have this implemented outside EngineCore (and then sending `abort` when stopping criteria?)

This might cause extra work. But it would enable fully async scheduling since the current inner loop is:

```python
scheduler.schedule()
executor.execute()
scheduler.update_from_output()
```

Reply:
cc @WoosukKwon